### PR TITLE
feat: add api resource CRD validation

### PR DESCRIPTION
**Pull request by kamiiiel**
see: https://gravitee.atlassian.net/browse/GKO-290

### CODE REVIEW

File: api/model/api/base/resource.go [hunk 1]
```diff
@@ -24,11 +24,11 @@ type Resource struct {
 	// Resource is enabled or not?
 	Enabled bool `json:"enabled"`
 	// Resource Name
-	Name string `json:"name,omitempty"`
+	Name string `json:"name"`
```
Comment (benoitgravitee) [inline api/model/api/base/resource.go:27]: I don't know how you usually proceed, but it deserve an entry in the release note / docs to make sure that users are fully aware of this as it is required now.

File: test/integration/admission/apiV4_create_withContext_and_invalid_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/gravitee-io/gravitee-kubernetes-operator/internal/errors"
+	"github.com/gravitee-io/gravitee-kubernetes-operator/test/internal/integration/assert"
+	"github.com/gravitee-io/gravitee-kubernetes-operator/test/internal/integration/constants"
+	"github.com/gravitee-io/gravitee-kubernetes-operator/test/internal/integration/fixture"
+	"github.com/gravitee-io/gravitee-kubernetes-operator/test/internal/integration/labels"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Validate update", labels.WithContext, func() {
```
Comment (benoitgravitee) [inline test/integration/admission/apiV4_create_withContext_and_invalid_resource_test.go:33]: type: create

File: test/integration/admission/apiV4_create_withContext_and_invalid_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	interval := constants.Interval
+	ctx := context.Background()
+	admissionCtrl := v4.AdmissionCtrl{}
+
+	It("should return error on API creation with invalid ai resource", func() {
```
Comment (benoitgravitee) [inline test/integration/admission/apiV4_create_withContext_and_invalid_resource_test.go:38]: "ai" ? typeo ?

File: test/integration/admission/apiV4_update_withContext_and_invalid_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Validate update", labels.WithContext, func() {
+	interval := constants.Interval
+	ctx := context.Background()
+	admissionCtrl := v4.AdmissionCtrl{}
+
+	It("should return error on API creation with invalid ai resource", func() {
```
Comment (benoitgravitee) [inline test/integration/admission/apiV4_update_withContext_and_invalid_resource_test.go:38]: type: "update" and "ai"

**Review by a-cordier — APPROVED** [review]

File: internal/admission/api/v4/validate.go [hunk 5]
```diff
@@ -95,6 +95,10 @@ func validateDryRun(ctx context.Context, api custom.ApiDefinitionResource) *erro
 		errs.AddSevere("unable to call dry run import because api is not a v4 API")
 	}
 
+	if impl.DefinitionContext == nil {
+		impl.DefinitionContext = v4.NewDefaultKubernetesContext()
```
Comment (a-cordier) [inline internal/admission/api/v4/validate.go:99]: I think this should be removed in favour of https://github.com/gravitee-io/gravitee-kubernetes-operator/pull/767

(At some point, we might have specific validation for syncFrom being MANAGEMENT)

**Pull request merged**